### ISSUE TITLE
Add an option to pause crawling

### ISSUE THREAD

**Issue by fawaf** [mwmbl/crawler-extension#25]
add option to disable crawling (e.g. the extension) in case of slowof cpu, slow network, other reason, etc.

**Comment by daoudclarke** [comment]
Also requested in a review: https://addons.mozilla.org/en-GB/android/addon/mwmbl-web-crawler/reviews/1853830/

**Comment by daoudclarke** [comment]
Since it is likely the user will forget to turn it back on (I know I will) I suggest we allow pausing, but not disabling. We could have two buttons:
 1. **Pause for +1h**. Pressing the button displays the message "Paused until xx:xx". Pressing repeatedly will increment the number of hours it's paused for.
 2. **Resume**. Pressing this button removes the message and resumes crawling.

If they want to turn it off completely they can do that through the Firefox extension UI.

**Comment by daoudclarke** [comment]
Behaviour for batches in progress: we need to check if pause has been pressed in the middle of processing a batch. If that is the case, then we should send the batch in its current state (even if there are just a few pages crawled) and stop crawling.

**Comment by fawaf** [comment]
err, i meant pausing mainly yeah, heh. maybe allow for a configurable option insead of just a "pause for +1 hour" button? a dropdown or even a direct textbox

**Comment by daoudclarke** [comment]
Personally I'd prefer to keep it simple, but I'll leave it to whoever implements it to decide.

**Comment by kristianpaul** [comment]
yeah, please add this, CPU was killing my laptop (i5,16G) recently and had to disable extension manually… Sorry guys

**Comment by xplosionmind** [comment]
It would be great to implement a simple keyboard shortcut to toggle crawling on and off (for exampbe `cmd` + `shift` + `M`).

**Comment by benob** [comment]
Another use case: disable on metered connections.
